typed/remove_test.go

```diff
@@ -0,0 +1,369 @@
+          scalar: string
+      - name: b
+        type:
+          map:
+            fields:
+            - name: c
+              type:
+                scalar: string
+            elementRelationship: separable
+    elementType:
+      scalar: string
+`,
+			remove: fieldpath.NewSet(fieldpath.MakePathOrDie("b")),
+			expect: `{"a": "value"}`,
+		},
+		{
+			object: `{"a": "value", "b": {"c":"value"}}`,
+			schema: `types:
+- name: type
+  map:
+    fields:
+      - name: a
+        type:
+          scalar: string
+      - name: b
+        type:
+          map:
+            fields:
+            - name: c
+              type:
+                scalar: string
+            elementRelationship: atomic
+    elementType:
+      scalar: string
+`,
+			remove: fieldpath.NewSet(fieldpath.MakePathOrDie("b", "c")),
+			expect: `{"a": "value", "b": {}}`,
+		},
+		{
+			object: `{"a": "value", "b":{"c":"value"}}`,
+			schema: `types:
+- name: type
+  map:
+    fields:
+      - name: a
+        type:
+          scalar: string
+      - name: b
+        type:
+          map:
+            fields:
+            - name: c
+              type:
+                scalar: string
+            elementRelationship: atomic
+    elementType:
+      scalar: string
+`,
+			remove: fieldpath.NewSet(fieldpath.MakePathOrDie("b")),
+			expect: `{"a": "value"}`,
+		},
+		{
+			object: `{"a": "value", "b": ["c"]}`,
+			schema: `types:
+- name: type
+  map:
+    fields:
+      - name: a
+        type:
+          scalar: string
+      - name: b
+        type:
+          list:
+            elementType:
+              scalar: string
+            elementRelationship: associative
+    elementType:
+      scalar: string
+`,
+			remove: fieldpath.NewSet(fieldpath.MakePathOrDie("b", "c")),
+			expect: `{"a": "value", "b": ["c"]}`,
+		},
+		{
+			object: `{"a": "value", "b": ["c"]}`,
+			schema: `types:
+- name: type
+  map:
+    fields:
+      - name: a
+        type:
+          scalar: string
+      - name: b
+        type:
+          list:
+            elementType:
+              scalar: string
+            elementRelationship: associative
+    elementType:
+      scalar: string
+`,
+			remove: fieldpath.NewSet(fieldpath.MakePathOrDie("b")),
+			expect: `{"a": "value"}`,
+		},
+		{
+			object: `{"a": "value", "b": [{"name": "c"}]}`,
+			schema: `types:
+- name: type
+  map:
+    fields:
+      - name: a
+        type:
+          scalar: string
+      - name: b
+        type:
+          list:
+            keys: 
+            - name
+            elementType:
+              map:
+                fields:
+                - name: name
+                  type:
+                    scalar: string
+                elementRelationship: atomic
+            elementRelationship: associative
+    elementType:
+      scalar: string
+`,
+			remove: fieldpath.NewSet(fieldpath.MakePathOrDie("b", fieldpath.PathElement{Key: &value.FieldList{{Name: "name", Value: value.NewValueInterface("c")}}})),
```

Comment:
OK I'm happy you have this example. There is a better way to build this though:
```suggestion
			remove: _NS(_P("b", _KBF("name", "c"))),
```

And as you can see, you can use these function aliases to simplify things throughout this file.

Also maybe you should have more than one item and show that only this one gets removed?

Reply:
Also maybe add other fields, so that we can see that the entire things gets removed.